src/main/java/edu/wisc/my/personalizedredirection/service/RedirectionServiceImpl.java

```diff
@@ -39,7 +38,8 @@ public String getUrl(HttpServletRequest request, UrlDataSource dataSource) throw
 			IRedirectURLSourceDataParser parser = getParser(dataSource);
 			mapList = parser.parseResource(resource);
 		} else {
-			throw new PersonalizedRedirectionException("Location " + locale + " not found.");
+		    logger.error("Invalid data location = " + dataSource.getDataSourceLocation());
```

Comment:
Somewhat a question of style, but I'd use `locale` here rather than a novel invocation of `dataSource.getDataSourceLocation()`, just in case somehow those don't turn out to be the same thing, since `locale` is already in scope (and is more precisely the thing that didn't exist.)